foliage.cabal

```diff
@@ -68,3 +69,22 @@ executable foliage
         vector               >=0.13.0.0   && <0.14,
         with-utf8            >=1.0.2.3    && <1.1,
         zlib                 >=0.6.2.3    && <0.7,
+
+test-suite foliage-test-suite
+  import:              common-settings
+  main-is:             Tests.hs
+  other-modules:       Foliage.Tests.Utils
+                     , Foliage.Tests.Tar
+  build-depends:       base
+                     , bytestring
+                     , containers
+                     , directory
+                     , filepath
+                     , network
+                     , process
+                     , tasty
+                     , tasty-hunit
+                     , tar
+                     , unix
+  build-tool-depends:  foliage:foliage
```

Comment:
If it were me, I'd probably split out the library at this point, have this depend on the library, and just call the foliage "main" function or whatever from the tests. It just avoids having another process call, and means your invocation of `foliage` is type-checked. But this is also fine.

Reply:
I am keen to test the program from the user prospective. We can still do what you suggest if we want to add unit tests.